### PR TITLE
Fixed race condition when getting fields

### PR DESCRIPTION
**Pull request by xperiandri**

### CODE REVIEW

File: RepoDb.Core/RepoDb/Cachers/DbFieldCache.cs [hunk 1]
```diff
@@ -13,6 +16,7 @@ namespace RepoDb
     public static class DbFieldCache
     {
         private static readonly ConcurrentDictionary<long, DbFieldCollection> cache = new();
+        private static readonly ConcurrentDictionary<IDbHelper, SemaphoreSlim> semaphores = new();
```
Comment (xperiandri) [inline RepoDb.Core/RepoDb/Cachers/DbFieldCache.cs:19]: `IDbHelper` may be different if you use several DBMSs at the same time like MS SQL Server and PostgreSQL

File: RepoDb.Core/RepoDb/Cachers/DbFieldCache.cs [hunk 2]
```diff
@@ -101,11 +105,31 @@ public static class DbFieldCache
...
             if (cache.TryGetValue(key, out var result) == false)
             {
-                // Get from DB
-                result = new DbFieldCollection(connection
-                    .GetDbHelper()
-                    .GetFields(connection, tableName, transaction),
-                    connection.GetDbSetting());
+                var dbHelper = connection.GetDbHelper();
+                var semaphore = semaphores.GetOrAdd(dbHelper, _ => new SemaphoreSlim(1, 1));
+                Debug.WriteLine($"Getting fields sync for '{tableName}' from {connection.GetType()}");
```
Comment (kavya-medcharla) [inline RepoDb.Core/RepoDb/Cachers/DbFieldCache.cs:110]: Is this for debugging purpose?

**Review by kavya-medcharla — APPROVED** [review]

File: RepoDb.Core/RepoDb/Cachers/DbFieldCache.cs [hunk 3]
```diff
@@ -101,11 +105,31 @@ public static class DbFieldCache
...
+                    else
+                    {
+                        Debug.WriteLine($"Fields for '{tableName}' from {connection.GetType()} already there");
+                    }
+                }
+                finally
+                {
+                    semaphore.Release();
+                    Debug.WriteLine($"Lock released to get fields sync for '{tableName}' from {connection.GetType()}");
+                }
```
Comment (andriilula) [inline RepoDb.Core/RepoDb/Cachers/DbFieldCache.cs:132]: This code is almost the same for [the async version](https://github.com/Lula-Technologies-Inc/RepoDB/pull/1/files#diff-bf9b3903d324d3a769bc83fb7845af7863fd477f8d34f6ca55f5b3bf7dae8d4eR218-R241). We should extract this into a function for a better code re-use

File: RepoDb.Core/RepoDb/Cachers/DbFieldCache.cs (hunk 2 above)
Comment (xperiandri) [reply RepoDb.Core/RepoDb/Cachers/DbFieldCache.cs:110]: Yes

**Pull request merged**